src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -3101,7 +3103,12 @@ NetworkOPsImp::transJson(
     transResultInfo(result, sToken, sHuman);
 
     jvObj[jss::type] = "transaction";
-    jvObj[jss::transaction] = transaction->getJson(JsonOptions::none);
+    // NOTE jvObj which is not a finished object for either API version. After
+    // it's populated, we need to finish it for a specific API version. This is
+    // done in a loop, near the end of this function.
+    std::string hash = {};
```

Comment:
I feel it would be more clear if we separate the serialization logic and data initialization in the `getJson()` function. `hash` can be directly initialized here as `hash = transaction->getTransactionID();` (instead of passing an optional ref-wrapper)

This is not a strong opinion, an alternative design suggestion, that's all

Reply:
Maybe. Trouble is that we do not have flexible design choice here, since `Transaction::getJson` is used in quite a few locations.

Reply:
Actually, having seen your (good, but unfortunately breaking API version 1) idea to remove `ledger_index` from inside transaction JSON output, I will try to do that properly both for `ledger_index` and for transaction `hash`. Hopefully in this PR, but unsure as this might come with larger churn that I am want here.

Reply:
ok done! It wasn't so bad - see the latest commit https://github.com/XRPLF/rippled/pull/4775/commits/95b6055ac487910b3eb2ba5d6194b94c751042b0